addons/mrp/models/mrp_bom.py

```diff
@@ -207,19 +211,25 @@ def _get_product2bom(self, products, bom_type=False):
         boms = self.search(domain, order='sequence, product_id')
         template2bom = {}
         variant2bom = {}
+        # Cannot use bom.sequence as unset sequence values are set to 0.
+        # Use counter to maintain the order of self.search above.
+        psql_sequence = 0
```

Comment:
It should work but
IMO, it is become a complicate (a bit messy) code. Maybe back-port the batch version of `_bom_find` (https://github.com/odoo/odoo/commit/974c4398b4eb3bb9fa097632e5223bd9efb6811c) in this method from master which it covers the case of order in account. What do you think?

Reply:
Yeah sure, it does look neater. I didn't know there was a new batch version of _bom_find starting 14.3 😮